src/AwsClient.php

```diff
@@ -490,6 +497,11 @@ private function addSignatureMiddleware(array $args)
                 $region = $signingRegionSet
                     ?? $commandSigningRegionSet
                     ?? $region;
+
+                MetricsBuilder::appendMetricsCaptureMiddleware(
```

Comment:
question: This function is invoked during the `sign` step.  How is the signature version being passed to the metrics builder by appending middleware to the build step, which has already completed?

Reply:
Good catch. Metrics here will not be collected. I will do a refactor for this type of cases.